File: road-trip.py
import sys

def main():
    with open(sys.argv[1]) as input_file:
        for line in input_file:
            cities = line.strip().split(';')[:-1]
            distances = []
            for city in cities:
                distances.append(int(city.split(',')[1]))
            
            last = 0
            result = []
            for d in sorted(distances):
                result.append(str(d-last))
                last = d

            print(','.join(result))

if __name__ == '__main__':
    main()

File: road-trip.lua
for line in io.lines(arg[1]) do
    distances = {}
    for m in line:gmatch('%d+') do distances[#distances+1] = tonumber(m) end
    table.sort(distances)

    last = 0
    result = {}
    for _, val in pairs(distances) do
        result[#result+1] = val - last
        last = val
    end
    
    print(table.concat(result, ','))
end
